composer/modules/server/services/ballerina-parser/src/main/java/org/ballerinalang/composer/service/ballerina/parser/service/BallerinaParserService.java

```diff
@@ -474,9 +484,7 @@ private JsonObject validateAndParse(BFile bFileRequest) throws InvocationTargetE
             JsonElement packageInfo = gson.toJsonTree(packageInfoJson.get());
             result.add("packageInfo", packageInfo);
         }
-        if (programDir != null) {
-            result.addProperty("programDirPath", programDir);
-        }
+        result.addProperty("programDirPath", programDir);
```

Comment:
This should be conditional, we prompt to open the program directory if this parameter is set.